### DISCUSSION TITLE
Transforming to Typescript: Add "with" properties

### DISCUSSION THREAD

**Discussion by ahoiroman** [spatie/laravel-data#148] [Q&A]
Hello everybody,

I am using `with()` to attach a list of permissions to the response sent by Laravel-data. Of course, these information are being lost on `php artisan typescript:transform`.

Is there any way to add those information to `generated.d.ts`? Something like:

```
   /** @typescript add permissions **/
    public function __construct(
        public Lazy|int            $id,
        public Carbon              $created_at,
        public ?Carbon             $deleted_at,
    ) {}

**Comment by rubenvanassche** [comment]
At the moment no such thing is possible since this would require extending types in TS transformer. We could accomplish such a thing with a custom attribute, which could be read by the data TS transformer. But I think adding a more structural way into TS transformer  3 to extend types would be a better solution. Added it to my list for things to consider in TypeScript Transformer 3!
